folly/Chrono.h

```diff
@@ -175,28 +175,57 @@ namespace folly {
 namespace chrono {
 
 struct coarse_steady_clock {
-  using rep = std::chrono::milliseconds::rep;
-  using period = std::chrono::milliseconds::period;
-  using duration = std::chrono::duration<rep, period>;
-  using time_point = std::chrono::time_point<coarse_steady_clock, duration>;
+  using duration = std::chrono::milliseconds;
+  using rep = duration::rep;
+  using period = duration::period;
+  using time_point = std::chrono::time_point<std::chrono::steady_clock, duration>;
```

Comment:
It seems odd for `some_clock::time_point::clock` to be different from `some_clock`. Do we have any examples of this being general practice?

Reply:
I've been thinking about it for a long time. This corresponds to the idea of the chrono library. If there is a guarantee that *_COARCE watches are the same watches as their fine-grained counterparts, then they simply must have time points pamametrized by one type of watch. It would be strange to use time_point_cast for example to compare the time points of the same clock.

Reply:
The question is whether these are the same clock even when they're different types. It makes more sense to me from a low-level perspective to treat them as different clocks but then to add a concept of compatible sets of clocks, where, within a compatible set of clocks, time-points of one clock are convertible to equivalent time-points of another clock. That way it remains possible to distinguish where it is necessary to distinguish and to treat as equivalent where equivalence makes sense.

Reply:
This concept should be introduced in the c++standard. In the meantime, what do we do if we just want to compare the time of these two hours? (except to compare their time_since_epoch)

Reply:
Perhaps it should exist in the standard. It should help to have an implementation when suggesting for something to exist in the standard though.

In the meantime, the concept of clock compatibility sets, with corresponding conversion functions, can exist in folly since the clocks being proposed are in folly.

Reply:
I agree. I removed the commit with this change from review